src/absa/training/atae_lstm.py

```
@@ -0,0 +1,38 @@
+
+def train_atae_lstm(train_rows, test_rows, *, epochs: int = 8, batch_size: int = 64, seed: int = 42):
+    torch.manual_seed(seed)
+    splits = split_official_data(train_rows, test_rows, seed=seed)
+    vocab = build_vocab([row.review_raw for row in splits.train])
+    model = ATAELSTM(len(vocab))
+    loader = DataLoader(TensorDataset(encode([r.review_raw for r in splits.train], vocab), encode([r.aspect for r in splits.train], vocab, 12), torch.tensor([LABEL_TO_ID[r.label] for r in splits.train])), batch_size=batch_size, shuffle=True)
+    optimiser = torch.optim.Adam(model.parameters(), lr=1e-3)
+    for _ in range(epochs):
+        model.train()
+        for review, aspect, label in loader:
+            optimiser.zero_grad(); loss=torch.nn.CrossEntropyLoss()(model(review, aspect), label); loss.backward(); optimiser.step()
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)